app/lib/app/SecvisogramPage/View/WizzardEditorTab/WizardPanel/editors/GenericEditor/ArrayEditor.js

```diff
@@ -38,27 +28,91 @@ export default function ArrayEditor({ property, instancePath }) {
   return (
     <>
       <div className="border-l border-r border-solid bg-gray-50 border-gray-400 wizard-menu-shadow mr-2">
-        <ul>
-          {sanitizedValue.map((_, i) => {
-            const indexErrors = errors.filter(
-              (e) => e.instancePath.startsWith('/' + [...instancePath, i].join('/'))
-            )
-            return (
-              <li key={instancePath.concat([String(i)]).join('.')}
-                  className={`${selectedIndex === i
-                    ? 'border-l-4 border-blue-400 border-b border-gray-300'
+        <Menu
+          value={sanitizedValue}
+          instancePath={instancePath}
+          property={property}
+        />
+      </div>
+      {selectedSubPath.length ? (
+        <GenericEditor
+          key={selectedPath.join('.')}
+          parentProperty={property}
+          property={childProperty}
+          instancePath={selectedPath}
+        />
+      ) : null}
+    </>
+  )
+}
+
+/**
+ * @param {object} props
+ * @param {Array<unknown>} props.value
+ * @param {string[]} props.instancePath
+ * @param {import('../../shared/types').Property} props.property
+ * @param {number} [props.level]
+ * @returns
+ */
+function Menu({ instancePath, level = 1, ...props }) {
+  const { property } = props
+
+  const { errors } = React.useContext(DocumentEditorContext)
+  const { selectedPath, setSelectedPath } = React.useContext(WizardContext)
+  const { setSideBarIsOpen, setSideBarSelectedPath } =
+    React.useContext(SideBarContext)
+  const { doc, updateDoc } = React.useContext(DocumentEditorContext)
+
+  const value = instancePath.reduce((value, pathSegment) => {
+    return (value ?? {})[pathSegment]
+  }, /** @type {Record<string, any> | null} */ (doc))
+  const sanitizedValue = Array.isArray(value) ? value : []
+  const recursionProperty =
+    property.metaInfo.arrayType?.metaInfo.propertyList?.find(
+      (p) => p.type === 'RECURSION'
+    )
+  const selectedPathSegment = selectedPath.slice(instancePath.length).at(0)
+  const selectedIndex = selectedPathSegment ? Number(selectedPathSegment) : null
+  const childProperty =
+    /** @type {NonNullable<typeof property.metaInfo.arrayType>} */ (
+      property.metaInfo.arrayType
+    )
+
+  /**
+   * @param {number} index
+   */
+  const setSelectedIndex = (index) => {
+    setSelectedPath(instancePath.concat([String(index)]))
+  }
+
+  return (
+    <div>
+      <ul className={`block ${level > 1 ? 'ml-2' : ''}`}>
+        {sanitizedValue.map((_, i) => {
+          const indexErrors = errors.filter((e) =>
+            e.instancePath.startsWith('/' + [...instancePath, i].join('/'))
+          )
+          return (
+            <li key={instancePath.concat([String(i)]).join('.')}>
+              <div
+                className={`${
+                  selectedIndex === i
```

Comment:
From the 3rd level on this seems to "select" all items on the same level in the hierarchy throughout the tree and I am confused as to why that happens. I'm not sure if this explains it well, so please see the attached screenshot:

![image](https://user-images.githubusercontent.com/8581199/198243990-87e190da-4abf-44c5-910f-0e053208ebbb.png)